File: LeetCode/Medium/3-longest-substring-without-repeating-characters/solution.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int low = 0, high = 0;
        HashMap<Character, Integer> f = new HashMap<>();
        int n = s.length();
        int res = 0;
        for (high = 0; high < n; high++) {
            char ch = s.charAt(high);
            f.put(ch, f.getOrDefault(ch, 0) + 1);
            while (f.size() < high - low + 1) {
                char left = s.charAt(low);
                f.put(left, f.get(left) - 1);
                if (f.get(left) == 0) {
                    f.remove(left);
                }
                low++;
            }
           int len = high - low + 1;
            res = Math.max(res, len);
        }
        return res;
    }
}

File: LeetCode/Medium/3-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        int low = 0, high = 0;
        unordered_map<char,int>f;
        int res = 0;
        for(high = 0;high<n;high++){
            f[s[high]]++;
            
            while(f.size()<high-low+1){
                f[s[low]]--;
                if(f[s[low]]==0){
                    f.erase(s[low]);
                }
                low++;
            }
           int len = high-low+1;
            res = max(len,res);
        }
        return res;
    }
};